pcdswidgets/vacuum/base.py

```diff
@@ -441,8 +454,62 @@ def _handle_icon_click(self):
         self._expert_display = display
         display.destroyed.connect(self._cleanup_expert_display)
 
-        if display:
-            display.show()
+        if len(self.ui_file_path) > 0:
+            self.tabWidget = QTabWidget()
+            self.embeddedDisplay = list()
+            self.tabWidget.setTabPosition(2)
+            self.tabWidget.addTab(display, "Typhos")
+
+            for i, files in enumerate(self.ui_file_path):
+                embedded = PyDMEmbeddedDisplay()
+
+                if i >= len(self.ui_file_titles):
+                    title = files
+                else:
+                    title = self.ui_file_titles[i]
+
+                if i >= len(self.ui_file_macros):
+                    macros = ''
+                else:
+                    macros = self.ui_file_macros[i]
```

Comment:
There's an, optional, more compact way to express this that you should know about using `itertools.zip_longest`:
```suggestion
            for files, title, macros in itertools.zip_longest(
                self.ui_file_path,
                self.ui_file_title,
                self.ui_file_macros,
            ):
                embedded = PyDMEmbeddedDisplay()
                title = title or files
                macros = macros or ''
```